src/libs/actions/PersonalDetails.js

```diff
@@ -285,6 +285,15 @@ function updateProfile(firstName, lastName, pronouns, timezone) {
                     }),
                 },
             },
+        }, {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: ONYXKEYS.FORMS.PROFILE_SETTINGS_FORM,
+            value: {isLoading: true},
+        }],
+        successData: [{
```

Comment:
I agree that pattern A is the correct one! So we should only include `optimisticData`, but no loading state.

Reply:
Is there a change for me here? Just confirming.

Reply:
Updated the `updateProfile` function as suggested.

> You'll also need to remove the Onyx key - PROFILE_SETTINGS_FORM

`PROFILE_SETTINGS_FORM` is [used inside](https://github.com/liyamahendra/Expensify/blob/a059ebe305b9f9df3bfc782a8cd4409940b95277/src/pages/settings/Profile/ProfilePage.js#L236) the `ProfilePage`

Reply:
We should still use the CONST and not directly input the key value, if that’s what you are suggesting @rushatgabhane

Reply:
I'm confused now @rushatgabhane - `PROFILE_SETTINGS_FORM` is defined in CONST. How do you suggest we remove the key and yet use it?

Reply:
Oh! I'm sorry @rushatgabhane, I totally misunderstood what you said in your initial comment. Somehow I was seeing that the PROFILE_SETTINGS_FORM was in CONST (must be because of lack of sleep). I just as per your suggestion :)